react/tic-tac-toe/src/app/screens/Players/index.js

```diff
@@ -1,13 +1,93 @@
-import React, { Fragment } from 'react';
+import React, { Fragment, Component } from 'react';
+import { connect } from 'react-redux';
+import { func, arrayOf } from 'prop-types';
 
+import getMatches from '../../../services/MatchesService';
+import actionsCreators from '../../../redux/match/actions';
 import TopBar from '../Game/components/TopBar';
+import { matchPropType } from '../../../constants/propTypes';
+import { PLAYER_ONE, TIE } from '../../../constants/gameConstants';
+import { history } from '../../../redux/store';
 
-function Players() {
-  return (
-    <Fragment>
-      <TopBar />
-    </Fragment>
-  );
+import styles from './styles.module.scss';
+
+
+class Players extends Component {
+  componentDidMount() {
+    if (this.props.matchHistory.length === 0) {
+      const { getMatchHistory } = this.props;
+      getMatches()
+        .then(response => getMatchHistory(response));
+    }
+  }
+
+  getPlayersMatches() {
+    const players = [{ won: 0, tie: 0, lost: 0 }, { won: 0, tie: 0, lost: 0 }];
+    this.props.matchHistory.forEach((match) => {
+      if (match.winner === TIE) {
```

Comment:
I'd create an auxiliary function that gets the winner, so you can directly do something like
```
const { winner, loser } = getResults(match);
if (!winner) players.forEach(player => player.tie++);
players[winner].won ++;
players[loser}.lost++;
```